src/Symfony/Component/Workflow/Marking.php

```diff
@@ -19,31 +19,54 @@
 class Marking
 {
     private $places = [];
-    private $context = null;
+    private $context;
 
     /**
      * @param int[] $representation Keys are the place name and values should be 1
      */
     public function __construct(array $representation = [])
     {
         foreach ($representation as $place => $nbToken) {
-            $this->mark($place);
+            $this->mark($place, $nbToken);
         }
     }
 
-    public function mark(string $place)
+    public function mark(string $place, int $nbToken = 1)
     {
-        $this->places[$place] = 1;
+        if ($nbToken < 1) {
+            throw new \LogicException(sprintf('The number of tokens must be greater than 0, "%s" given.', $nbToken));
+        }
+
+        if (!\array_key_exists($place, $this->places)) {
+            $this->places[$place] = 0;
+        }
+        $this->places[$place] += $nbToken;
     }
 
-    public function unmark(string $place)
+    public function unmark(string $place, int $nbToken = 1)
```

Comment:
this change is a BC break for child classes overriding the method with extra parameter(s). It should rather be done the smooth way on 7.1